### DISCUSSION TITLE
getServerSession with minimum authOptions

### DISCUSSION THREAD

**Discussion by lveillard** [nextauthjs/next-auth#8965] [Help]
### Summary

So we have 3 nextjs apps and only one does the authentication, we rewrite all the /auth requests to the 3rd one.

The problem is that one of our apps works with the appDir and we want to make getServerSession work in a server component. 

However to make it work we need to pass it the authOptions which are in the 3rd app

It's ok, we can do that or a shared lib, but i'm curious about alternatives.

- Could we use getSession in RSC? (in my tests the result is null :S)
- Could we create minimal version of the authOptions enought to make the get ServerSession work?

So far i discovered:
- The secret is not needed if it is in the .env
- prividers are not necessary they actually get overrided
- debug is used but not a key thing
- adapter: ❓ Not sure if it is required as a param inside the authOptions for the getServersession to work. I guess this one is needed
- page: : ❓ Same, but it does not look like relevant to get the session
- callbacks: : ❓No idea, i guess not?

### Additional information

```js
Btw for the redirections we just did something like this in the authOptions


async redirect({ url, baseUrl }) {
	console.log('url', url, 'baseUrl', baseUrl);
	// Check if state parameter exists in the URL
	if (url.includes('originApp=myapp1')) {
		return `${'localhost:3000'}${url.replace(baseUrl, '')}`; //replace by process.env.APP1_URL or similar
	}
	if (url.includes('originApp=myapp2')) {
		return `${'localhost:3001'}${url.replace(baseUrl, '')}`; 
	}
	// Fallback to the default behavior
	return baseUrl;
		},
```

**Comment by lveillard** [comment]
Btw to make the rewrites work i had to remove NEXTAUTH_URL because it overrides my redirects from :3002 to :3001 and :3000.

Without NEXTAUTH_URL, i'm able to redirect to the original next auth apps after login in with the providers

**Comment by lvegerano** [comment]
I'm having the same issue. Why would this be different?
